cypress/integration/data-processor.spec.js

```diff
@@ -188,7 +188,8 @@ describe("Box selection", () => {
     );
 
     cy.wrap(dataProcessor)
-      .should("have.property", "index")
+      .should("have.property", "specificationHelper")
+      .then(() => dataProcessor.indexDataIfNotAlreadyIndexed())
```

Comment:
One hacky way I've done testing of cached operations is to do something like this:

```
dataProcessor.expensiveOperationThatIsCached();
for(let i = 0; i < 1000; i++) { // If it's not cached, this will take very long
  dataProcessor.expensiveOperationThatIsCached(); // Should return immediately. 
}
```